apis/v1/secretproviderclass_types.go

```diff
@@ -46,13 +46,32 @@ type SecretObject struct {
 	Data        []*SecretObjectData `json:"data,omitempty"`
 }
 
+type Secret struct {
+	// name of the object to sync
+	ObjectName string `json:"objectName,omitempty"`
+	// expected format of the secret received from the provider (e.g. plaintext, json)
+	Format string `json:"format,omitempty"`
+	// JSON path to target for a secret received in JSON format
+	JsonPath string `json:"jsonPath,omitempty"`
+}
+
+type TransformOptions struct {
```

Comment:
I think we actually want the `ProviderFilePath` or `InFilePath` here to note the input data for the translation. I could see a single `SecretProviderClass` that has a number of different secrets that you want to split out in different ways.